### PR TITLE
Run karma tests on saucelabs (fixes #60)

### LINKED ISSUE #60: Run karma tests on saucelabs

### PR DESCRIPTION
**Pull request by muffinresearch**
In the end I decided rather than leaking the access key I'd make it work so that:

* PRS from 3rd party repos (including our own) will run the tests against FF only. 
* PRs on the main repo will run all the tests on saucelabs
* Commits on the main repo will run all test tests on saucelabs.

In other words this would mean for our current work flow SauceLabs would only be used when stuff lands (e.g. sort of a CI approach). If we need to run sauce on a PR then you can as a team member push your branch to `mozilla/payments-ui` rather than your fork and then you'll get the full saucelabs tests on your pr (obviously this requires you to have the commit bit on our repo).

### CODE REVIEW

**Comment by muffinresearch** [comment]
I've sent a question to Saucelabs re: whether they are OK with people publishing the access key to workaround needing to push branches to the main repo like this.

**Comment by muffinresearch** [comment]
See https://github.com/mozilla/payments-ui/pull/185 to see that a standard PR from a different repo runs just a normal test suite run in FF.

File: README.md [hunk 1]
```diff
@@ -44,7 +47,33 @@ autocmd FileType javascript let b:syntastic_checkers = findfile('.eslintrc', '.;
 
 ### Tests
 
-To run the tests run: `grunt test`
+To run the tests locally run: `grunt test`. This will run the unit tests
+against Firefox.
+
+#### Cross-browser testing
+
+The tests are run only on Firefox when a PR is submitted. When that code is landed
```
Comment (kumar303) [inline README.md]: maybe expand 'landed' here to 'landed on master'

File: karma.conf.js [hunk 2]
```diff
@@ -1,49 +1,7 @@
...
-// that we don't want for tests.
-delete webpackConfig.output;
-delete webpackConfig.entry;
-
-// Add this just for testing:
-webpackConfig.plugins = [
-  new RewirePlugin(),
-];
-
+var karmaConfig = require('./karma.config.shared');
```
Comment (kumar303) [inline karma.conf.js]: same comment here: ``karma.conf.shared`` might be a bit more consistent (just a nit)

File: tests/test.complete-payment.jsx [hunk 3]
```diff
@@ -11,21 +11,24 @@ var helpers = require('./helpers');
...
+      },
+    };
     React = require('react');
     TestUtils = require('react/lib/ReactTestUtils');
-
     this.CompletePayment = TestUtils.renderIntoDocument(
       <CompletePayment productId='mozilla-concrete-brick'
-                       userEmail={this.email} />
+                       userEmail={this.email}
+                       win={this.win} />
```
Comment (kumar303) [inline tests/test.complete-payment.jsx]: well, hey, that works !

**Comment by kumar303** [comment]
r+wc

**Pull request merged**